Engine_Revit_UI/Create/Workset.cs

```diff
@@ -0,0 +1,72 @@
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.UI.Revit.Engine
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Creates a new workset with a specified name.")]
+        [Input("document", "Revit model file.")]
+        [Input("worksetName", "Name of the new workset to be added.")]
+        [Output("workset", "Existing workset if it exists or a newly created workset.")]
+        public static Workset Workset(Document document, string worksetName)
+        {
+            // Skip non-workshared documents because they can't contain worksets
+            if (document.IsWorkshared == false)
+                return null;
+
+            // Skip null or empty workset names            
+            if (string.IsNullOrWhiteSpace(worksetName))
+                return null;
+
+            // Avoid recreating a workset if one with the same name already exists
+            FilteredWorksetCollector worksets = new FilteredWorksetCollector(document).OfKind(WorksetKind.UserWorkset); // Find all user worksets
+            Workset workset = worksets.Where(x => x.Name == worksetName).FirstOrDefault(); // Find the specified workset if it exists
+            if (workset != null)
+            {
+                // The specified workset already exists
+                return workset; // Return the existing workset
```

Comment:
Is it worth adding a warning to users to let them know they have been given an existing workset due to the name existing rather than a new one?

Reply:
Warning added as suggested

Reply:
I would disagree with @FraserGreenroyd here. This warning will trigger every time an object is pushed/updated in Revit. I would rather add a warning when new workset is added - worksets are a very sensitive aspect of the model and are usually managed by the BIM manager. Adding a few by coincidence is the last thing that should happen, while using the existing workset is perfectly fine.

Reply:
Hmm by that logic @pawelbaran this wouldn't be a create method, if create is the edge case that we want to avoid? I would imagine a user (I would be at least) being confused if they used a create method and got a warning saying "hey, you just created an object, are you sure?" - I'd be like "yeah, course I'm sure, I asked for it" as opposed to using a 'create' method to get an in-use workset.

Based on your comment @pawelbaran it might be better to be a Query method instead, looking up an existing workset and returning a new one if one does not exist?

Reply:
Hmm, I would say it is a `Create` method as such @FraserGreenroyd, assuming that `Query` should not change the document, but only query it. However, I do agree that recording warning on expected behaviour would be hm... eccentric 🐒

On the other hand, warning on creation of a new workset and no warning on reusing the existing one is definitely the desirable end-user behaviour. This means we have 2 use cases with 2 different warnings:
- Attempted creation of a workset from scratch using `Create` method - desired warning when the existing workset is reused
- This case, i.e. setting workset param and warning the user if that action required creation of a new one

They are pretty much contradicting. Now, thinking about that a bit more, I would dare to say that creation of worksets is such a rare and sensitive case (max. few times in the entire project life?) that I would rather block it from happening by coincidence (on push of a single element). So, I lean towards checking if a workset exists under a given name, and if not then warning and no action. `Create` method disappears then and gets reimplemlemented as `Convert` once we have a BHoM wrapper for it that could be pushed.

What do you think @AndreBalasian?

Reply:
I suggest having both a Query and a Create method.
The Query method would be called first and return the required workset if it exists, or trigger a Create method if it doesn't.

Reply:
As for the warning message, let's leave that out of the Query method and only produce a warning through the Create method when the Query method calls it.

Reply:
Again, I believe it is dangerous from the project point of view. After a deeper thought, I cannot imagine pushing worksets somewhat as an attachement to single elements - this seems to be completely against general Revit policy and BIM standards. Elements should be driven by worksets, not the opposite.

So to allow creating Revit worksets from BHoM, I would strongly insist on creating a wrapper in Revit_oM and correspondent `Convert` method in Engine_Revit_UI. By these means we will make sure that the worksets will get created only on purpose (which I believe should be the case).

Reply:
Following a chat with @pawelbaran the Create method has been converted to a Query method that returns an existing workset with a specified name if it exists and null if it doesn't.
No new worksets will be created by this routine if they don't already exist.